src/View/Helper/FormHelper.php

```diff
@@ -2672,4 +2675,29 @@ public function implementedEvents()
     {
         return [];
     }
+
+    public function getValuesSources()
+    {
+        return $this->_valuesSources;
+    }
+
+    public function setValuesSources($sources)
+    {
+        $this->_valuesSources = array_values(array_intersect((array)$sources, ['context', 'data', 'query']));
+
+        return $this;
+    }
+
+    public function getSourceValue($fieldname)
+    {
+        foreach ($this->getValuesSources() as $valuesSource) {
+            if ($valuesSource === 'context') {
+                return $this->_getContext()->val($fieldname);
+            }
+            if (isset($this->request->{$valuesSource}[$fieldname])) {
```

Comment:
> So how would I differentiate between a legitimate null/false and the method just not finding anything within the path?

You don't need to since post data and query string can only have string values.